### PR TITLE
Add test cases for compiler crash in from-clause with do-clause

### LINKED ISSUE #36622: Compiler crash in from-clause with do-clause
**Description:**
```
function returnNil() {
}

function bar() returns string|error {
    check from int _ in [1, 3, 5]
    do {        
        check returnNil();
        return "str";
    };
    return "";
}

public function main() returns error? {
    _ = check bar();
}
```
This example crashes in master.

**Affected Versions:**
Ballerina 2201.1.0 (Swan Lake)

### PR DESCRIPTION
**Pull request by SasinduDilshara**
Fixes #36622

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by SasinduDilshara** [comment]
We need to wait until https://github.com/ballerina-platform/ballerina-lang/pull/36647 got revert for master branch.
Currently https://github.com/ballerina-platform/ballerina-lang/pull/36647 got revert for [ballerina-platform:2201.2.0-stage](https://github.com/ballerina-platform/ballerina-lang/tree/2201.2.0-stage) branch only. https://github.com/ballerina-platform/ballerina-lang/pull/37108

File: tests/jballerina-unit-test/src/test/resources/test-src/query/query-action.bal [hunk 1]
```diff
@@ -69,6 +69,39 @@ function testSimpleQueryAction2() returns int{
     return count;
 }
 
+function testSimpleQueryAction3() returns error? {
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/query/query-action.bal:72]: Do these tests actually test the scenario in https://github.com/ballerina-platform/ballerina-lang/issues/36622? I tried running these on 2201.1.0 and it didn't seem to crash whereas the code in the issue caused a crash.

Comment (SasinduDilshara) [reply tests/jballerina-unit-test/src/test/resources/test-src/query/query-action.bal:72]: I will check

Comment (SasinduDilshara) [reply tests/jballerina-unit-test/src/test/resources/test-src/query/query-action.bal:72]: Updated. a54ee512c764cf67e5cd64d991a28448439772a2

**Review by KavinduZoysa — APPROVED** [review]

**Pull request merged**